pandas/__init__.py

```diff
@@ -212,6 +212,7 @@
 Pyarrow will become a required dependency of pandas in the next major release of pandas (pandas 3.0),
 (to allow more performant data types, such as the Arrow string type, and better interoperability with other libraries)
 but {pa_msg}
+To resolve this warning, please specify the pyarrow extra during installation: pandas[pyarrow]
```

Comment:
Let's hold on this PR for now. 

There's a decently high likelihood that we end up reverting this warning altogether.
(I know it's super frustrating to have the warning - I'm hoping to reach a conclusion at the next dev meeting and then release a hopefully warning-free pandas 2.1.1)